### PR TITLE
3 authentication

### LINKED ISSUE #71: AaD, I want to redirect the user to the dashboard after signup/login

### PR DESCRIPTION
**Pull request by rsanchez-dv**
Resolves #71 user Dashboard.

### CODE REVIEW

**Comment by afterafx** [comment]
I get an error when trying to run this branch locally. 

Module not found: Can't resolve 'jwt-decode'

File: src/App.js [hunk 1]
```diff
@@ -27,54 +29,99 @@ const links = {
...
+            path={links.login}
+            isSomething={!this.state.isAuthenticated}
+            redirectLink={links.dashboard}
+            component={RegistrationPage}
+            handleLogin={this.handleLogin}
+          />
+          {/* Dashboard Protected Route */}
+          <ProtectedRoute
+            path={links.dashboard}
+            isSomething={this.state.isAuthenticated}
```
Comment (emily-yang) [inline src/App.js]: @devmarantz wtf is this `isSomething` attribute? Did you mean `isAuthenticated`?

File: src/components/NavBar.js [hunk 2]
```diff
@@ -88,10 +93,15 @@ class NavBar extends React.Component {
...
-            <NavLink onClick={() => this.closeMenu()} to={links.dashboard} />
+            {this.state.isAuthenticated ? (
+              <NavLink onClick={() => this.closeMenu()} to={links.dashboard}>
+              Dashboard
+              </NavLink>
+            ) : (null
+            )}
+            {this.state.isAuthenticated ? (<NavLink onClick={()=>{console.log('Logging OFF'); this.props.handleLogoff();}} to="/">Log Off</NavLink>) : (
+              <NavLink to={links.login}>Login</NavLink>)}
+
```
Comment (emily-yang) [inline src/components/NavBar.js]: Mobile links for authentication should be moved to the bottom to stay consistent with the order for normal web app

File: src/components/pages/DashboardPage.js [hunk 3]
```diff
@@ -1,7 +1,74 @@
...
+
+  goToOverview = () => {
+    this.setState({ active: 'overview' });
+  };
+
+  goToSettings = () => {
+    this.setState({ active: 'settings' });
+  };
+
+  // TODO: Implement Logout
```
Comment (emily-yang) [inline src/components/pages/DashboardPage.js]: Logout logic moved to nav bar

File: src/components/registration/SignUp.js [hunk 4]
```diff
@@ -20,36 +21,16 @@ class SignUp extends Component {
...
-      // Modal
-      showModal: false,
-      // Shows error message if server fails
-      showRegistrationFailure: false,
-      // Shows when waiting for server response
-      showSpinner: false,
+      // Showsgdfgsdgss error message if server fails
+      showRegsgdfgsdgsistrationFailure: false,
+      // Showsgdfgsdgss when waiting for server response
+      showSpisgdfgsdgsnner: false,
```
Comment (emily-yang) [inline src/components/registration/SignUp.js]: Who did this

**Review by emily-yang — CHANGES_REQUESTED** [review]
Please see comments in the PR

File: src/App.js (hunk 1 above)
Comment (devmarantz) [reply src/App.js]: Yeah I did!  Changed it, thank you for catching this.

File: src/components/pages/DashboardPage.js [hunk 5]
```diff
@@ -1,7 +1,78 @@
...
+          showSettings={this.goToSettings}
+          logout={this.logout}
+        />
+        {this.state.active === 'overview' ? (
+          <div className="dashboardBody">
+            <h1 className="dashboardHeader">
+Welcome,
+              {' '}
+              {this.state.userName ? this.state.userName : 'Anon'}
+!
```
Comment (emily-yang) [inline src/components/pages/DashboardPage.js]: Suggestion: use a template string
```
{`Welcome, ${this.state.userName || 'Anon'}`}

```

File: src/components/pages/DashboardPage.js [hunk 6]
```diff
@@ -1,7 +1,78 @@
...
+          <div className="dashboardBody">
+            <h1 className="dashboardHeader">
+              {this.state.userName}
+'s Account
+            </h1>
+            <p className="dashboardText">
+              {' '}
+User ID:
+              {' '}
+              {this.state.userID}
```
Comment (emily-yang) [inline src/components/pages/DashboardPage.js]: Suggestion: use a template string

**Review by emily-yang — APPROVED** [review]

File: src/components/registration/SignUp.js [hunk 7]
```diff
@@ -208,7 +183,9 @@ class SignUp extends Component {
           />
           <p>
               By clicking "Sign Up" you are agreeing
-              to our <a href="www.google.com">Terms and Agreement</a>
+              to our
+            {' '}
+            <a href="www.google.com">Terms and Agreement</a>
```
Comment (emily-yang) [inline src/components/registration/SignUp.js]: This link goes to `http://localhost:3000/www.google.com`

File: src/components/NavBar.js [hunk 8]
```diff
@@ -35,7 +36,8 @@ class NavBar extends React.Component {
 
   render() {
     const { menuOpen } = this.state;
-    const { links } = this.props;
+    const { links, isAuthenticated } = this.props;
+    this.state.isAuthenticated = isAuthenticated;
```
Comment (emily-yang) [inline src/components/NavBar.js]: Never mutate state directly. Alternative: 
```
  componentDidUpdate(prevProps, prevState) {
    const { isAuthenticated } = this.props;
    if (isAuthenticated !== prevState.isAuthenticated) {
      this.setState({ isAuthenticated });
    }
  }
```

**Review by emily-yang — CHANGES_REQUESTED** [review]

File: src/components/NavBar.js (hunk 8 above)
Comment (rsanchez-dv) [reply src/components/NavBar.js]: @devmarantz You did this section

File: src/components/registration/SignUp.js [hunk 9]
```diff
@@ -208,7 +183,9 @@ class SignUp extends Component {
           />
           <p>
               By clicking "Sign Up" you are agreeing
-              to our <a href="www.google.com">Terms and Agreement</a>
+              to our
+            {' '}
+            <a href="https://www.termsandcondiitionssample.com/live.php?token=bYAxBa2kby8ugr9x4eWMbKKgXnxOQyNg" target="_blank">Terms and Agreement</a>
```
Comment (emily-yang) [inline src/components/registration/SignUp.js]: Add `rel="noopener noreferrer"` to the tag to prevent security vulnerability

**Review by emily-yang — CHANGES_REQUESTED** [review]

**Review by devmarantz — APPROVED** [review]
Resolved requested changes

**Review by emily-yang — APPROVED** [review]

**Pull request merged**